winlogbeat/eventlog/eventlog.go

```diff
@@ -82,39 +84,63 @@ type Record struct {
 func (e Record) ToEvent() beat.Event {
 	m := common.MapStr{
 		"type":          e.API,
-		"log_name":      e.Channel,
-		"source_name":   e.Provider.Name,
-		"computer_name": e.Computer,
 		"record_number": strconv.FormatUint(e.RecordID, 10),
-		"event_id":      e.EventIdentifier.ID,
 	}
 
+	m.Put("event.kind", "event")
+	m.Put("event.dataset", fmt.Sprintf("%v.%v", e.API, strings.ToLower(e.Channel)))
```

Comment:
Experiment. Not really distinct datasets in the `module.fileset` of other beats. But here's what the values would look like, in the real world:

Modern windows:

- wineventlog.application
- wineventlog.security
- wineventlog.system
- ...

Vista and older, iirc:

- eventlogging.application
- eventlogging.security
- eventlogging.system
- ...

Reply:
"eventlogging" is a really old (before Vista) limited format for these events. Has way less information than "wineventlog". However afaik "wineventlog" was strictly additions vs "eventlogging".

Comparing between application & security, for example, I'm not sure. Probably somewhat possible (just like you can somewhat visualize apache.access & .error together with some caveats), when you have the same source name.

I understand the semantics are different from what we do elsewhere, as the other places the two positions here are really detailing which source this comes from: `[source short name].`[type of log generated by the source]`.

But unfortunately in Windows event logs, the closest thing resembling a source name is the provider name, which is really verbose (e.g. "Microsoft-Windows-Security-Auditing").

Perhaps we should just take the `event.dataset` out for now, and add later, after taking some time to consider how to do this right. The information itself is still present under winlog.*

Reply:
++ on not using event.dataset for this as it's not consitent with other use cases.

Reply:
Ok I answered this from a very different angle than you were asking, actually.

The winlog.* fields except the content of event_data or user_data should always be almost identical, as it's mostly metadata from the win event log pipeline.

Reply:
Ok, removing.

Reply:
The way I read it: There is just one `event.dataset` if we even populate it. It could be `event.dataset: winlogevent` or `winlog.event` or similar. We can still add it later.

Reply:
Not quite. It's more like the bullet lists I had pasted above. E.g. recent windows, you'd have a combination of:

* wineventlog.application
* wineventlog.security
* wineventlog.system
* ...

But the "wineventlog" part never changes.